2004-02-26  Richard Earnshaw  <rearnsha@arm.com>
	Daniel Jacobowitz  <drow@mvista.com>

	PR target/14302
	* arm.h (ARM_GO_IF_LEGITIMATE_ADDRESS): Don't check the mode
	size for minipool references.

Index: arm.h
===================================================================
RCS file: /cvs/gcc/gcc/gcc/config/arm/arm.h,v
retrieving revision 1.168.2.1
retrieving revision 1.168.2.2
diff -u -r1.168.2.1 -r1.168.2.2
--- gcc/config/arm/arm.h	25 Feb 2004 11:50:24 -0000	1.168.2.1
+++ gcc/config/arm/arm.h	26 Feb 2004 14:10:01 -0000	1.168.2.2
@@ -2029,7 +2029,7 @@
 	   && GET_CODE (XEXP (X, 0)) == REG				\
 	   && ARM_REG_OK_FOR_BASE_P (XEXP (X, 0)))			\
     goto LABEL;								\
-  else if (GET_MODE_SIZE (MODE) >= 4 && reload_completed		\
+  else if (reload_completed						\
 	   && (GET_CODE (X) == LABEL_REF				\
 	       || (GET_CODE (X) == CONST				\
 		   && GET_CODE (XEXP ((X), 0)) == PLUS			\
